c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -1492,12 +1523,50 @@ c2_status_t MfxC2DecoderComponent::AllocateFrame(MfxC2FrameOut* frame_out)
                     /* Buffer locked, try next block. */
                     MFX_DEBUG_TRACE_PRINTF("Buffer still locked, try next block");
                     res = C2_TIMED_OUT;
-                }
-                else
+                } else {
                     *frame_out = MfxC2FrameOut(std::move(out_block), it->second);
+                }
             }
         } else {
-            res = MfxC2FrameOut::Create(converter, out_block, m_mfxVideoParams.mfx.FrameInfo, TIMEOUT_NS, frame_out, hndl.get());
+            if (m_mfxVideoParams.mfx.FrameInfo.Width >= WIDTH_2K || m_mfxVideoParams.mfx.FrameInfo.Height >= HEIGHT_2K) {
+                // Thumbnail generation for 4K/8K video
+                if (m_surfacePool.size() < m_surfaceNum) {
```

Comment:
The m_surfaces didn't limit count, m_surfacePool need to limit?

Reply:
Actually m_surfaces is fixed. m_surfaces.size() is equal to response->NumFrameActual https://github.com/projectceladon/MediaSDK_C2/blob/master/c2_utils/src/mfx_va_frame_pool_allocator.cpp#L140.